consumers/writers/api/logging.go

```diff
@@ -38,5 +38,21 @@ func (lm *loggingMiddleware) Consume(msgs interface{}) (err error) {
 		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
 	}(time.Now())
 
-	return lm.consumer.Consume(msgs)
+	return lm.consumer.ConsumeBlocking(msgs)
+}
+
+func (lm *loggingMiddleware) Consume(msgs interface{}, errs chan<- error) {
+	ch := make(chan error)
+	var err error
+	defer func(begin time.Time) {
+		message := fmt.Sprintf("Method consume took %s to complete", time.Since(begin))
+		if err != nil {
+			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			return
+		}
+		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+	}(time.Now())
```

Comment:
This method **is blocking** and `consumer.Consume` will practically never (successfully) end, so logging this way does not make sense.

Reply:
I have updated these in previous commit